scripts/jenkins/cloud/ansible/deploy-caasp-v4-terraform.yml

```diff
@@ -0,0 +1,568 @@
+      ssh -o StrictHostKeyChecking=no sles@{{item}} sudo update-ca-certificates
+    with_items: "{{ master_ips.value.values() | list }}"
+
+  - name: Management Machine | Copy certificates to each of the worker nodes in the cluster
+    shell: |
+      set -e
+      scp -o StrictHostKeyChecking=no {{ ca_path }} sles@{{item}}:~/ca-bundle.pem
+      ssh -o StrictHostKeyChecking=no sles@{{item}}  sudo cp ~/ca-bundle.pem {{ ca_path }}
+      ssh -o StrictHostKeyChecking=no sles@{{item}} sudo update-ca-certificates
+    with_items: "{{ worker_ips.value.values() | list }}"
+
+  - name: Management Machine | Run skuba init
+    shell: |
+      set -e
+      cd {{ caasp_openstack_path }}
+      {{ skuba_init_command }} | tee -a skuba_init.log
+    register:
+      skuba_init_command_results
+    failed_when: " not ('[init] configuration files written to' in skuba_init_command_results.stdout)"
+
+  - name: Management Machine | Move openstack.conf
+    shell: |
+      set -e
+      cd {{ caasp_openstack_path }}
+      {{ skuba_mv_openstack_conf_command }}
+    register:
+      skuba_mv_openstack_conf_command_results
+
+  - name: Management Machine | Run skuba bootstrap
+    shell: |
+      set -e
+      eval `ssh-agent -s`
+      ssh-add
+      cd {{ caasp_openstack_path }}/{{stack_name}}-cluster
+      {{ item }} | tee -a skuba_bootstrap.log
+    with_items: "{{ skuba_boostrap_commands }}"
+    register:
+      skuba_bootstrap_commands_results
+    failed_when: " not ('[bootstrap] successfully bootstrapped core add-ons' in skuba_bootstrap_commands_results.stdout)"
+
+  - name: Set expected total nodes in the cluster
+    set_fact:
+      total: "{{ num_masters | int + num_workers | int}}"
+
+  - name: Set expected total nodes joining the cluster after bootstrap
+    set_fact:
+      total_join: "{{ num_masters | int + num_workers | int - 1}}"
+
+  - name: Management Machine | Run skuba join
+    shell: |
+      set -e
+      eval `ssh-agent -s`
+      ssh-add
+      cd {{ caasp_openstack_path }}/{{stack_name}}-cluster
+      {{ item }} | tee -a skuba_join.log
+    with_items: "{{ skuba_join_commands }}"
+    register:
+      skuba_join_commands_results
+
+  - name: Management Machine | track the count of the nodes that join the cluster successfully
+    set_fact:
+      count: "{{ count | default(0) | int + 1}}"
+    when: "'node successfully joined the cluster' in item"
+    with_items: "{{ skuba_join_commands_results.results|map(attribute='stdout')|list }}"
+
+  - name: Management Machine | Fail when the expected number of nodes do not join the cluster
+    fail:
+      msg: "Not all expected nodes have successfully joined the cluster"
+    when: "count | int != total_join | int"
+
+  - name: Management Machine | Run skuba cluster status to validate all the nodes are in the cluster in Ready state
+    shell: |
+      set -e
+      cd {{ caasp_openstack_path }}/{{stack_name}}-cluster
```

Comment:
keeping the same pattern for variables, here `stack_name` does not have white spaces before/after `{{ }}`